### PR TITLE
GH-107 add mime type and property filters

### LINKED ISSUE #107: Support for Xray Mime Type and Property filters for Watch
**Is your feature request related to a problem? Please describe.**

Xray Watch is missing the Mime Type and Property filters. https://www.jfrog.com/confluence/display/JFROG/Configuring+Xray+Watches#ConfiguringXrayWatches-Step3SetFiltersonRepositories

Screenshot 2023-02-28 at 8 59 17 AM

**Describe the solution you'd like**

Add support for both types of filter to watch resource. e.g.

```hcl
resource "xray_watch" "mime_type_property" {
  name   = "mime-property-example"
  active = true

  watch_resource {
    type = "repository"

    filter {
      type  = "mime-type"
      value = "application/json"
    }

    kv_filter {
      type  = "property"
      value {
        key = "prop-name"
        value = "prop-value"
    }
  }
}
```

**Describe alternatives you've considered**

N/A

**Additional context**

Include support for import

### PR DESCRIPTION
**Pull request by alexhung**
Closes #107 

- Add `mine-type` string filter
- Add new Key/Value filter `kv_filter` type to support `property` filter.

### CODE REVIEW

File: docs/resources/watch.md [hunk 1]
```diff
@@ -77,6 +77,11 @@ resource "xray_watch" "repository" {
       type  = "regex"
       value = ".*"
     }
+
+    filter {
+      type  = "mine-type"
```
Comment (shrajfr12) [inline docs/resources/watch.md:82]: Should be mime-type right ?

File: examples/resources/xray_watch/resource.tf [hunk 2]
```diff
@@ -59,6 +59,11 @@ resource "xray_watch" "repository" {
       type  = "regex"
       value = ".*"
     }
+
+    filter {
+      type  = "mine-type"
```
Comment (shrajfr12) [inline examples/resources/xray_watch/resource.tf:64]: mime-type instead of mine-type ?

**Review by shrajfr12 — COMMENTED** [review]
found some typos with mime-type (spelled as mine-types)

File: examples/resources/xray_watch/resource.tf (hunk 2 above)
Comment (alexhung) [reply examples/resources/xray_watch/resource.tf:64]: Good catch!

**Review by shrajfr12 — APPROVED** [review]

File: pkg/xray/watches.go [hunk 3]
```diff
@@ -255,6 +289,20 @@ func packAntFilter(filter WatchFilter) (map[string]interface{}, error) {
...
+	}
+
+	return map[string]interface{}{
+		"type":  filter.Type,
+		"key":   kvValue.Key,
+		"value": kvValue.Value,
+	}, nil
+}
+
 var packFilterMap = map[string]map[string]interface{}{
```
Comment (danielmkn) [inline pkg/xray/watches.go:306]: 8-0

**Review by danielmkn — APPROVED** [review]

**Pull request merged**